src/olympia/scanners/models.py

```diff
@@ -192,7 +193,10 @@ class AbstractScannerRule(ModelBase):
         max_length=200,
         help_text=_('This is the exact name of the rule used by a scanner.'),
     )
-    scanner = models.PositiveSmallIntegerField(choices=SCANNERS.items())
+    select_scanner = SCANNERS
+    del select_scanner[WAT]
+    del select_scanner[MAD]
+    scanner = models.PositiveSmallIntegerField(choices=select_scanner.items())
```

Comment:
You can't do this - `select_scanner = SCANNERS` doesn't create a copy of `SCANNERS`, you're modifying the original dict, which will affect all other code that references the same constants (and because the server instances are reused, other requests/users too).  

A generator function would work e.g. `[name: scanner for name, scanner in SCANNERS.items() if name not in [WAT, MAD]]`.  Anyway though, changing the model is different to changing `formfield_for_choice_field` like @willdurand suggested - would this work @willdurand or does the model need to stay as it is?

Reply:
IMHO the model need to stay as it is. Those scanners _are_ valid choices, we just don't want them to show up in the rule creation at the moment.

Reply:
> would this work @willdurand or does the model need to stay as it is?

No, as @diox said, we only want to hide these choices in the form but they are still valid choices so we shouldn't change the model.